bindings/swc_cli/src/commands/compile.rs

```diff
@@ -380,77 +346,133 @@ impl CompileOptions {
+                        .write(map.as_bytes())
                         .and(Ok(()))?;
                 }
 
-                buf.write(r.code.as_bytes()).and(Ok(()))
+                buf.write(transform_output.code.as_bytes()).and(Ok(()))
             })?;
 
-            buf.flush()
-                .context("Failed to write output into single file")
+            buf.flush().context("Failed to write output")
         } else {
             inputs.into_par_iter().try_for_each(
                 |InputContext {
                      compiler,
                      fm,
-                     options,
+                     mut options,
                      file_path,
                      file_extension,
                  }| {
-                    let result = execute(compiler, fm, options);
-
-                    match result {
-                        Ok(output) => {
-                            emit_output(&output, &self.out_dir, &file_path, file_extension)
+                    let out_file = if let Some(ref out_dir) = &self.out_dir {
+                        Some(resolve_output_file_path(
+                            out_dir,
+                            &file_path,
+                            file_extension,
+                        )?)
+                    } else {
+                        None
+                    };
+
+                    // Create a source map path
+                    let map_file = if let Some(ref out_file) = out_file {
+                        Some(create_map_file(out_file))
+                    } else {
+                        None
+                    };
+
+                    // When the `SourceMapsConfig::Linked` is instatiated, it contains
+                    // an empty path that needs to be updated dynamically
+                    if options.source_maps == Some(SourceMapsConfig::Linked(PathBuf::new())) {
+                        // Override the initial path
+                        options.source_maps =
+                            Some(SourceMapsConfig::Linked(map_file.clone().unwrap()));
+                    };
+
+                    // Transform the input
+                    match execute(compiler, fm, options) {
+                        Ok(transform_output) => {
+                            if let Some(ref out_file) = out_file {
+                                let out_dir =
+                                    out_file.parent().expect("Parent should be available");
+
+                                if !out_dir.is_dir() {
+                                    fs::create_dir_all(out_dir)?;
+                                }
+
+                                fs::write(&out_file, &transform_output.code)?;
+
+                                if let Some(map) = &transform_output.map {
+                                    fs::write(map_file.unwrap(), map)?;
+                                }
+                            } else {
+                                println!(
```

Comment:
Seems like a mistake?

Reply:
Previously, the name of the source map was defined in the aforementioned function. But this does not work anymore. It is now required to know the name of the source map beforehand. Therefore, I moved this function and made it part of the executor.

Reply:
The function still exists, it is just inside of the executor.